horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/TestServiceNoRestTest.java

```diff
@@ -0,0 +1,504 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@QuarkusTest
+@QuarkusTestResource(PostgresResource.class)
+@TestProfile(HorreumTestProfile.class)
+@TestTransaction
+@TestSecurity(user = DEFAULT_USER, roles = {Roles.TESTER, Roles.VIEWER, FOO_TEAM, FOO_TESTER})
+class TestServiceNoRestTest extends BaseServiceNoRestTest {
+
+   @Inject
+   TestService testService;
+
+   @org.junit.jupiter.api.Test
+   void testCreateTest() {
+      Test t1 = createSampleTest("test", null, null, null);
+      Test t2 = createSampleTest("1234", null, null, null);
+
+      Test created1 = testService.add(t1);
+      assertNotNull(created1.id);
+      assertEquals(1, TestDAO.count());
+      TestDAO test1 = TestDAO.findById(created1.id);
+      assertEquals(t1.name, test1.name);
+      assertEquals(FOO_TEAM, test1.owner);
+      assertEquals(Access.PUBLIC, test1.access);
+      assertEquals(true, test1.notificationsEnabled);
+      assertNull(test1.folder);
+
+      Test created2 = testService.add(t2);
+      assertNotNull(created2.id);
+      assertEquals(2, TestDAO.count());
+      TestDAO test2 = TestDAO.findById(created2.id);
+      assertEquals(t2.name, test2.name);
+      assertEquals(FOO_TEAM, test2.owner);
+      assertEquals(Access.PUBLIC, test2.access);
+      assertEquals(true, test2.notificationsEnabled);
+      assertNull(test2.folder);
+   }
+
+   @org.junit.jupiter.api.Test
+   void testCreateTestWithFolders() {
+      Test t1 = createSampleTest("test", null, "folder/trailing/slash/", null);
+      Test t2 = createSampleTest("1234", null, " folder/with/spaces  ", null);
+      Test t3 = createSampleTest("another-test", null, " folder/with/spaces/trailing/slash  /  ", null);
+      Test t4 = createSampleTest("single-slash", null, "  /  ", null);
+
+      TestDAO test1 = TestDAO.findById(testService.add(t1).id);
+      assertEquals(1, TestDAO.count());
+      assertEquals("folder/trailing/slash", test1.folder);
+
+      TestDAO test2 = TestDAO.findById(testService.add(t2).id);
+      assertEquals(2, TestDAO.count());
+      assertEquals("folder/with/spaces", test2.folder);
+
+      TestDAO test3 = TestDAO.findById(testService.add(t3).id);
+      assertEquals(3, TestDAO.count());
+      assertEquals("folder/with/spaces/trailing/slash", test3.folder);
+
+      TestDAO test4 = TestDAO.findById(testService.add(t4).id);
+      assertEquals(4, TestDAO.count());
+      assertNull(test4.folder);
+   }
+
+   @org.junit.jupiter.api.Test
+   void testCreateTestWithBlankName() {
+      Test t = createSampleTest("", null, null, null);
+
+      // empty test name
+      ServiceException thrown = assertThrows(ServiceException.class, () -> testService.add(t));
+      assertEquals("Test name can not be empty", thrown.getMessage());
```

Comment:
Same comment I made in `SchemaServiceNoRestTest` about testing the Response http code against expected code